apollo-router/src/plugins/connectors/http_json_transport.rs

```diff
@@ -272,89 +223,496 @@ pub(crate) enum HttpJsonTransportError {
+            make_uri(
+                None,
+                &template,
+                &map! {
+                    "b" => 4,
+                    // "c" => 5,
+                    "d" => 6,
+                    "x" => 1,
+                    "y" => 2,
+                    "z" => 3
+                }
+            )
+            .err()
+            .unwrap()
+            .to_string(),
+            r#"Could not generate path from inputs: Missing variable c for required parameter {b},{c};{d!} given variables {"b":4,"d":6,"x":1,"y":2,"z":3}"#
+        );
+
+        assert_eq!(
+            make_uri(
+                None,
+                &template,
+                &map! {
+                    // "b" => 4,
+                    // "c" => 5,
+                    "d" => 6,
+                    "x" => 1,
+                    "y" => 2,
+                    "z" => 3
+                }
+            )
+            .err()
+            .unwrap()
+            .to_string(),
+            r#"Could not generate path from inputs: Missing variable b for required parameter {b},{c};{d!} given variables {"d":6,"x":1,"y":2,"z":3}"#
+        );
+
+        let line_template = "http://localhost/line/{p1.x},{p1.y},{p1.z}/{p2.x},{p2.y},{p2.z}"
+            .parse()
+            .unwrap();
+
+        assert_eq!(
+            make_uri(
+                None,
+                &line_template,
+                &map! {
+                    "p1.x" => 1,
+                    "p1.y" => 2,
+                    "p1.z" => 3,
+                    "p2.x" => 4,
+                    "p2.y" => 5,
+                    "p2.z" => 6,
+                }
             )
             .unwrap()
             .as_str(),
-            "https://localhost:8080/v1/users/user:1"
+            "http://localhost/line/1,2,3/4,5,6"
+        );
+
+        assert_eq!(
+            make_uri(
+                None,
+                &line_template,
+                &map! {
+                    "p1.x" => 1,
+                    "p1.y" => 2,
+                    "p1.z" => 3,
+                    "p2.x" => 4,
+                    "p2.y" => 5
+                    // "p2.z" => 6,
+                }
+            )
+            .err()
+            .unwrap()
+            .to_string(),
+            r#"Could not generate path from inputs: Missing required variable p2.z in {"p1.x":1,"p1.y":2,"p1.z":3,"p2.x":4,"p2.y":5}"#
+        );
+
+        assert_eq!(
+            make_uri(
+                None,
+                &line_template,
+                &map! {
+                    "p1.x" => 1,
+                    // "p1.y" => 2,
+                    "p1.z" => 3,
+                    "p2.x" => 4,
+                    "p2.y" => 5,
+                    "p2.z" => 6
+                }
+            )
+            .err()
+            .unwrap()
+            .to_string(),
+            r#"Could not generate path from inputs: Missing required variable p1.y in {"p1.x":1,"p1.z":3,"p2.x":4,"p2.y":5,"p2.z":6}"#
```

Comment:
i'm wondering if this is a potential security issue — if these error messages are exposed, they could reveal information that should be kept internal to query planning.

at first i was thinking this won't be a problem, because we should be validating `$args` and `$this` to ensure they don't select things that won't be provided. but because we have support for `$config`, which we can't validate now, it's pretty easy to get this message in a client response:
```
  "errors": [
    {
      "message": "TransportError(TemplateGenerationError(\"Missing required variable $config.x in {\\\"$this.__typename\\\":\\\"Track\\\",\\\"$this.id\\\":\\\"c_0\\\"}\"))",
      "extensions": {
        "code": "FETCH_SERVICE"
      }
    }
  ],
```

Reply:
Maybe we should get rid of the context, then and just specify the name of the missing variable? I assume you can only see these errors in dev mode, right? Otherwise it'd be easy to leak secrets by showing the current context.

Reply:
I went ahead and removed it for now just in case. We can always add back more details if folks struggle to debug this.

Reply:
the errors become top-level graphql errors right now, so i think removing this data is the right move